### ISSUE TITLE
Floquet solver not working

### ISSUE THREAD

**Issue by clmntwg** [qutip/qutip#1207]
I try to run the code for finding floquet energies from the qutip user guide. 

```python
import numpy as np
from qutip import *
import matplotlib.pyplot as plt

delta = 0.2 * 2*np.pi; eps0 = 1.0 * 2*np.pi; A = 2.5 * 2*np.pi; omega = 1.0 * 2*np.pi
H0 = - delta/2.0 * sigmax() - eps0/2.0 * sigmaz()
H1 = A/2.0 * sigmaz()
args = {'w': omega}
H = [H0, [H1, 'sin(w * t)']]
T = 2*pi / omega
f_modes_0, f_energies = floquet_modes(H, T, args)
f_energies
```

 I got the following error 

```text
  File "<ipython-input-6-76f4206b44f0>", line 1, in <module>
    runfile('C:/Users/605912/OneDrive - BOOZ ALLEN HAMILTON/DARPA/John Burke/Quantum apertures/calculations/floquet.py', wdir='C:/Users/605912/OneDrive - BOOZ ALLEN HAMILTON/DARPA/John Burke/Quantum apertures/calculations')

  File "C:\Users\605912\AppData\Local\Continuum\anaconda3\lib\site-packages\spyder_kernels\customize\spydercustomize.py", line 827, in runfile
    execfile(filename, namespace)

  File "C:\Users\605912\AppData\Local\Continuum\anaconda3\lib\site-packages\spyder_kernels\customize\spydercustomize.py", line 110, in execfile
    exec(compile(f.read(), filename, 'exec'), namespace)

  File "C:/Users/605912/OneDrive - BOOZ ALLEN HAMILTON/DARPA/John Burke/Quantum apertures/calculations/floquet.py", line 27, in <module>
    f_modes_0, f_energies = floquet_modes(H, T, args)

  File "C:\Users\605912\AppData\Local\Continuum\anaconda3\lib\site-packages\qutip\floquet.py", line 102, in floquet_modes
    U = propagator(H, T, [], args)

  File "C:\Users\605912\AppData\Local\Continuum\anaconda3\lib\site-packages\qutip\propagator.py", line 186, in propagator
    _safe_mode=False)

  File "C:\Users\605912\AppData\Local\Continuum\anaconda3\lib\site-packages\qutip\sesolve.py", line 157, in sesolve
    ss = _sesolve_QobjEvo(H, tlist, args, options)

  File "C:\Users\605912\AppData\Local\Continuum\anaconda3\lib\site-packages\qutip\sesolve.py", line 189, in _sesolve_QobjEvo
    H_td.compile(omp=nthread)

  File "C:\Users\605912\AppData\Local\Continuum\anaconda3\lib\site-packages\qutip\qobjevo.py", line 1559, in compile
    self.tlist)

  File "C:\Users\605912\AppData\Local\Continuum\anaconda3\lib\site-packages\qutip\qobjevo_codegen.py", line 124, in _compiled_coeffs
    "CompiledStrCoeff", True)

  File "C:\Users\605912\AppData\Local\Continuum\anaconda3\lib\site-packages\qutip\qobjevo_codegen.py", line 82, in _import_str
    "tmpfile:" + try_file + ext) from e

UnboundLocalError: local variable 'e' referenced before assignment
```

**Comment by Ericgig** [comment]
String format coefficients (`'sin(w * t)' ` in H) which use cython for compilation often fails on windows. Using functions instead work better.
```
def f(t,args):
    return sin(args["w"]*t)
H = [H0, [H1, f]]
```
Here there is also an error in the error message, we will fix that bug soon.

**Comment by geophase** [comment]
Hi,  I think you may have some typos in your code, for example you have a pi instead of np.pi in line 9. Also in the definition of the constants you have 2np.pi instead of 2*np.pi. 

I've run your code changing those typos and I didn't get any error. I am running it in Mac OS X Catalina. 

Try to run the code I've run  in your machine: 

```
import numpy as np
from qutip import *
import matplotlib.pyplot as plt

delta = 0.2 * 2*np.pi; eps0 = 1.0 * 2*np.pi; A = 2.5 * 2*np.pi; omega = 1.0 * 2*np.pi
H0 = - delta/2.0 * sigmax() - eps0/2.0 * sigmaz()
H1 = A/2.0 * sigmaz()
args = {'w': omega}
H = [H0, [H1, 'sin(w * t)']]
T = 2*np.pi / omega
f_modes_0, f_energies = floquet_modes(H, T, args)
f_energies
```

**Comment by clmntwg** [comment]
> String format coefficients (`'sin(w * t)' ` in H) which use cython for compilation often fails on windows. Using functions instead work better.
> 
> ```
> def f(t,args):
>     return sin(args["w"]*t)
> H = [H0, [H1, f]]
> ```
> 
> Here there is also an error in the error message, we will fix that bug soon.


Thanks this worked.


> import numpy as np
> from qutip import *
> import matplotlib.pyplot as plt
> 
> delta = 0.2 * 2*np.pi; eps0 = 1.0 * 2*np.pi; A = 2.5 * 2*np.pi; omega = 1.0 * 2*np.pi
> H0 = - delta/2.0 * sigmax() - eps0/2.0 * sigmaz()
> H1 = A/2.0 * sigmaz()
> args = {'w': omega}
> H = [H0, [H1, 'sin(w * t)']]
> T = 2*np.pi / omega
> f_modes_0, f_energies = floquet_modes(H, T, args)
> f_energies

This didnt' work for me.  I'm also running it on a Mac

**Comment by jakelishman** [comment]
The particular reported bug was fixed in #1199, and released in QuTiP 4.5.2.
